server/controllers/authController.js

```diff
@@ -114,13 +114,52 @@ class AuthController {
               email,
               username,
               role,
+              lineManager,
               isVerified
             }
           }
         }));
       },
     )(req, res);
   }
+
+  /**
+     * This method handle the sign request.
+     * @param {object} req The user's request.
+     * @param {object} res The response.
+     * @returns {object} The status and some data of the user.
+     */
+  static async socialAuth(req, res) {
+    const {
+      id, firstname, lastname, email, role, isVerified, lineManager
+    } = req.user;
```

Comment:
I don’t see a generated username as well. By signing in with social platforms, If you don’t have access to the username, you should use their first name as their username.

Reply:
Feedback Implemented!